falcon/routing/compiled.py

```diff
@@ -26,6 +26,8 @@
 from falcon.util.misc import is_python_func
 from falcon.util.sync import _should_wrap_non_coroutines, wrap_sync_to_async
 
+if False:
+    from typing import Any
```

Comment:
As I understand, this is a [special workaround for Python 3.5.0 and 3.5.1](https://stackoverflow.com/a/57940134).
As such, it could warrant a `NOTE` for the reader.